pkg/vm/cli/cli.go

```diff
@@ -1,197 +1,316 @@
+		Func: handleXStack,
+	},
+	{
+		Name: "istack",
+		Help: "Show invocation stack contents",
+		LongHelp: "Show invocation stack contents",
+		Func: handleXStack,
+	},
+	{
+		Name: "loadavm",
+		Help: "Load an avm script into the VM",
+		LongHelp: `Usage: loadavm <file>
+<file> is mandatory parameter, example:
+> load /path/to/script.avm`,
+		Func: handleLoadAVM,
+	},
+	{
+		Name: "loadhex",
+		Help: "Load a hex-encoded script string into the VM",
+		LongHelp: `Usage: loadhex <string>
+<string> is mandatory parameter, example:
+> load 006166`,
+		Func: handleLoadHex,
+	},
+	{
+		Name: "loadgo",
+		Help: "Compile and load a Go file into the VM",
+		LongHelp: `Usage: loadhex <file>
+<file> is mandatory parameter, example:
+> load /path/to/file.go`,
+		Func: handleLoadGo,
+	},
+	{
+		Name: "run",
+		Help: "Execute the current loaded script",
+		LongHelp: `Usage: run [<operation> [<parameter>...]]
+
+<operation> is an operation name, passed as a first parameter to Main() (and it
+        can't be 'help' at the moment)
+<parameter> is a parameter (can be repeated multiple times) that can be specified
+        as <type>:<value>, where type can be:
+            'bool': supports 'false' and 'true' values
+            'int': supports integers as values
+            'string': supports strings as values (that are pushed as a byte array
+                      values to the stack)
+       or can be just <value>, for which the type will be detected automatically
+       following these rules: 'true' and 'false' are treated as respective
+       boolean values, everything that can be converted to integer is treated as
+       integer and everything else is treated like a string.
 
-var commands = map[string]command{
-	"help":    {0, "show available commands", false},
-	"exit":    {0, "exit the VM prompt", false},
-	"ip":      {0, "show the current instruction", true},
-	"break":   {1, "place a breakpoint (> break 1)", true},
-	"estack":  {0, "show evaluation stack details", false},
-	"astack":  {0, "show alt stack details", false},
-	"istack":  {0, "show invocation stack details", false},
-	"loadavm": {1, "load an avm script into the VM (> load /path/to/script.avm)", false},
-	"loadhex": {1, "load a hex string into the VM (> loadhex 006166 )", false},
-	"loadgo":  {1, "compile and load a .go file into the VM (> load /path/to/file.go)", false},
-	"run":     {0, "execute the current loaded script", true},
-	"cont":    {0, "continue execution of the current loaded script", true},
-	"step":    {0, "step (n) instruction in the program (> step 10)", true},
-	"ops":     {0, "show the opcodes of the current loaded program", true},
+Passing parameters without operation is not supported. Parameters are packed
+into array before they're passed to the script, so effectively 'run' only
+supports contracts with signatures like this:
+   func Main(operation string, args []interface{}) interface{}
+
+Example:
+> run put string:"Something to put"`,
+		Func: handleRun,
+	},
```

Comment:
opening bracket missing somewhere

Reply:
Nope, it compiles just fine.

Reply:
true, I see that is ending of previous part